assets/activation-scripts/activate.d/start.bash

```diff
@@ -17,20 +17,25 @@ export | LC_ALL=C $_coreutils/bin/sort > "$_start_env"
 # other things) prepending this environment's bin directory to the PATH.
 # shellcheck disable=SC2154 # set in the main `activate` script
 if [ -d "$_profile_d" ]; then
-  declare -a _profile_scripts
-  # TODO: figure out why this is needed
-  set +e
-  read -r -d '' -a _profile_scripts < <(
-    cd "$_profile_d" || exit
-    shopt -s nullglob
-    echo *.sh
-  )
-  set -e
-  for profile_script in "${_profile_scripts[@]}"; do
-    # shellcheck disable=SC1090 # from rendered environment
-    source "$_profile_d/$profile_script"
-  done
-  unset _profile_scripts
+  if [ "$MODE" == "run" ]; then
+    # shellcheck source-path=SCRIPTDIR/etc/profile.d
+    source "$_profile_d/0100_run-mode-paths.sh"
+  else
```

Comment:
(non-blocking for this PR but important for release) What's the level of confidence about not wanting to run the other `etc/profile.d` scripts?

Do we need a distinction between things that may be used for:
1. development, and known to leak with breakages, such as `PYTHON_PATH` described [here](https://www.notion.so/floxdev/007-Initial-pass-at-runtime-mode-12e744c8e31680a3b62df6f6b717805f?pvs=4#12e744c8e3168064bbffef20a84a18b7) and same for Rust
2. runtime, may be safer, and regress existing behaviour if they were missing such as CUDA and Jupyter

Are we going to cover off any of this with changes to documentation, such as suggesting that you should still use separate layered environments for development?

Reply:
> What's the level of confidence about not wanting to run the other etc/profile.d scripts?

I'd say 100%, because the entire raison d'être for their existence is to conjure up a replacement for the Nix build shell. But that is _not_ to say that we might want to have a separate set of scripts to invoke for hardware concerns or other site-specific considerations. I would just argue that those should be kept distinct from the existing scripts specifically so that we can address the requirement to invoke them separately as required based on the activation mode.